src/Service/DocumentService.php

```diff
@@ -0,0 +1,213 @@
+        $this->callService   = $callService;
+
+    }//end __construct()
+
+
+    /**
+     * Executes the prepare file request to xxllnc.
+     *
+     * The prepare file request is needed to send the inhoud of a enkelvoudiginformatieobject and retrieve a reference id to connect it to a xxllnc case.
+     *
+     * @param array  $infoObject
+     * @param Source $xxllncApi
+     *
+     * @return string|null $reference File reference.
+     */
+    private function prepareFile(array $infoObject, Source $xxllncApi)
+    {
+        if (isset($infoObject['informatieobject']['_self']['id']) === false) {
+            return null;
+        }
+
+        $infoObjectEntity = $this->entityManager->find('App:ObjectEntity', $infoObject['informatieobject']['_self']['id']);
+        if ($infoObjectEntity instanceof ObjectEntity === false) {
+            return null;
+        }
+
+        $base64 = $infoObjectEntity->getValueObject('inhoud')->getFiles()->first()->getBase64();
+        // $base64 = \Safe\base64_decode($infoObjectEntity->getValueObject('inhoud')->getFiles()->first()->getBase64());
+        $file_stream = $base64;
+        // $file_stream = Utils::streamFor($base64);
+        $multipart = [
+            [
+                'name'     => 'upload',
+                'contents' => $file_stream,
+                'filename' => $infoObject['informatieobject']['bestandsnaam'],
+            ],
+        ];
+
+        // Send the POST request to xxllnc.
+        var_dump('Request prepare file:');
+        $config = [
+            'headers'   => ['Content-Type' => 'multipart/form-data'],
+            'debug'     => true,
+            'multipart' => $multipart,
+        ];
+        // try {
+            $response = $this->callService->call($xxllncApi, '/case/prepare_file', 'POST', $config);
+            $result   = $this->callService->decodeResponse($xxllncApi, $response);
+            var_dump('Rerefence id: ', array_key_first($result['result']['instance']['references']));
+            $reference = array_key_first($result['result']['instance']['references']) ?? null;
+        // } catch (Exception $e) {
+        // var_dump($e->getMessage());
+        // return null;
+        // }//end try
+        return $reference ?? null;
+
+    }//end prepareFile()
+
+
+    /**
+     * Reserves a document number at the xxllnc api.
+     *
+     * A document number is needed to save the actual file/document/informatieobject.
+     *
+     * @param Source $xxllncApi
+     *
+     * @return string|false $documentNumber Document number else false.
+     */
+    private function reserveDocumentNumber(Source $xxllncApi)
+    {
+        // Send the POST request to xxllnc.
+        try {
+            $response = $this->callService->call($xxllncApi, '/document/reserve_number', 'POST');
+            $result   = $this->callService->decodeResponse($xxllncApi, $response);
+            var_dump('Document number: ', $result['result']['instance']['serial']);
```

Comment:
Var_dump

Reply:
yes this work is in progress and shouldn't be triggered

Reply:
Alright